src/emulator/auth/operations.ts

```diff
@@ -2800,6 +2994,276 @@ function updateTenant(
+    oauthIdToken?: string;
+    oauthAccessToken?: string;
+    oauthRefreshToken?: string;
+    oauthTokenSecret?: string;
+    oauthExpiresIn?: string;
+  } = {},
+  timeoutMs: number = 60000
+): Promise<{
+  updates: BlockingFunctionUpdates;
+  extraClaims?: Record<string, unknown>;
+}> {
+  const url = state.getBlockingFunctionUri(event);
+
+  // No-op if blocking function is not present
+  if (!url) {
+    return { updates: {} };
+  }
+
+  const jwt = generateBlockingFunctionJwt(state, event, url, timeoutMs, user, options, oauthTokens);
+  const reqBody = {
+    data: {
+      jwt,
+    },
+  };
+
+  const controller = new AbortController();
+  const timeout = setTimeout(() => {
+    controller.abort();
+  }, timeoutMs);
+
+  let response;
+  try {
+    const res = await fetch(url, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(reqBody),
+      signal: controller.signal,
+    });
+    const text = await res.text();
+    assert(
+      res.ok,
+      `BLOCKING_FUNCTION_ERROR_RESPONSE: ((HTTP request to ${url} returned an error: ${text}))`
+    );
+    response = JSON.parse(text) as BlockingFunctionResponsePayload;
+  } catch (thrown: any) {
+    const err = thrown instanceof Error ? thrown : new Error(thrown);
+    const isAbortError = err.name.includes("AbortError");
+    if (isAbortError) {
+      throw new InternalError(
+        `BLOCKING_FUNCTION_ERROR_RESPONSE: ((Deadline exceeded making request to ${url}.))`,
+        err.message
+      );
+    }
+    throw new InternalError(
+      `BLOCKING_FUNCTION_ERROR_RESPONSE: ((Failed to make request to ${url}.))`,
+      err.message
+    );
+  } finally {
+    clearTimeout(timeout);
+  }
+
+  return processBlockingFunctionResponse(response);
+}
+
+function processBlockingFunctionResponse(response: BlockingFunctionResponsePayload): {
+  updates: BlockingFunctionUpdates;
+  extraClaims?: Record<string, unknown>;
+} {
+  // Only return updates that are specified in the update mask
+  let extraClaims;
+  const updates: BlockingFunctionUpdates = {};
+  if (response.userRecord) {
+    const userRecord = response.userRecord;
+    assert(
+      userRecord.updateMask,
+      "BLOCKING_FUNCTION_ERROR_RESPONSE: ((Response UserRecord is missing updateMask.))"
+    );
+    const mask = userRecord.updateMask;
+    const fields = mask.split(",");
+
+    for (const field of fields) {
+      switch (field) {
+        case "displayName":
+        case "photoUrl":
+        case "disabled":
+        case "emailVerified":
+          (updates as any)[field] = userRecord[field];
+          break;
+        case "customClaims":
+          validateSerializedCustomClaims(userRecord.customClaims!);
+          updates.customAttributes = userRecord.customClaims;
+          break;
+        // Session claims are only returned in beforeSignIn. For more info,
+        // see https://cloud.google.com/identity-platform/docs/blocking-functions#modifying_a_user
```

Comment:
If a developer makes a mistake and returns `sessionClaims` from `beforeCreate`, the implementation will try to parse it instead of ignores it. Is there anyway we can address this?